src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -579,6 +579,7 @@ void ClassLoader::setup_module_search_path(JavaThread* current, const char* path
   new_entry = create_class_path_entry(current, path, &st,
                                       false /*is_boot_append */, false /* from_class_path_attr */);
   if (new_entry != nullptr) {
+    assert(new_entry->is_jar_file(), "module path entry %s is not a jar file", new_entry->name());
```

Comment:
How do we guarantee that new_entry is never a JAR file? Do we never come here if --module-path points to an exploded directory? A comment would be helpful.

Reply:
I've added the following comment:
  `// ClassLoaderExt::process_module_table() filters out non-jar entries before calling this function.`